packages/driver/src/multi-domain/serializer.ts

```diff
@@ -0,0 +1,39 @@
+import { isFunction, isSymbol } from 'lodash'
+
+/**
+ * serialize. Used for serializing subject to be returned from subdomains to the primary domain.
+ * @param value any object to be serialized.
+ * @returns the serialized object as a string or undefined, if the object cannot be serialized.
+ */
+const serialize = (value: any): string => {
+  let serializedSubject: string = undefined!
+
+  try {
+    serializedSubject = JSON.stringify(value, (key, value) => {
+      // If we encounter any unserializable values we want to abort the whole process.
+      if (isFunction(value) || isSymbol(value) || value === undefined) {
```

Comment:
We can't port `undefined` using json, even trying to sub it out for our own custom string would require us to write our own parse function. I could transform `undefined` values into `null` but that may be confusing.

Reply:
I think we probably need to discuss this with Ben and Brian. We talked about making unserialized values undefined, but not about objects that might have unserializable values within them, so I think we need to define that behavior. I'm not sure if throwing away the whole object is the right way to go or not.